doc/collision_detection.md

```diff
@@ -0,0 +1,212 @@
+      Vector2(1, 0),
+      Vector2(0, -1),
+      Vector2(-1, 0),
+    ]);
+    addShape(shape);
+  }
+}
+```
+
+The `HitboxPolygon` added to the `Collidable` here is a diamond shape(◇).
+More about how the different shapes are defined in the [Shapes](#Shapes) section.
+
+Remember that you can add as many `HitboxShape`s as you want to your `Collidable` to make up more
+complex hitboxes. For example a snowman with a hat could be represented by three `HitboxCircle`s and
+a `HitboxPolygon` as its hat.
+
+To react to a collision you should override the `collisionCallback` in your component.
+Example:
+
+```dart
+class MyCollidable extends PositionComponent with Hitbox, Collidable {
+  ...
+
+  @override
+  void onCollision(Set<Vector2> points, Collidable other) {
+    if (other is CollidableScreen) {
+      ...
+    } else if (other is YourOtherCollidable) {
+      ...
+    }
+  }
+}
+```
+
+In this example it can be seen how the Dart `is` keyword is used to check which other `Collidable`
+that your component collided with. The set of points is where the hitboxes of the collidables
+intersected. Note that the `onCollision` method will be called on both collidable components if they
+have both implemented the `onCollision` method.
+
+If you want to check collisions with the screen edges, as we do in the example above, you can use
+the predefined [ScreenCollidable](#ScreenCollidable) class and since that one aldo is a `Collidable`
+you can implement your own `onCollision` method for that class if needed.
+
+### HasCollidables
+If you want to use this collision detection in your game you have to add the `HasCollidables` mixin
+to your game so that the game knows that it should keep track of which components that can collide.
+
+Example:
+```dart
+class MyGame extends BaseGame with HasCollidables {
+  ...
+}
+```
+
+Now when you add your `Collidable` components to your game they will automatically be checked for
+collisions.
+
+### ScreenCollidable
+`ScreenCollidable` is not a mixin, but a pre-made collidable component which represents the edges of
+your game/screen. If you add a `ScreenCollidable` to your game your other collidables will be
+notified when they collide with the edges. It doesn't take any arguments, it only depends on the
+`size` of the game that it is added to. To add it you can just do `add(ScreenCollidable())` in
+your game, if you don't want the `ScreenCollidable` itself to be notified when something collides
+with it. If you so want it to be notified, then you need to extend it and implement the
+`collisionCallback` method for it.
+
+## Shapes
+### Shape
+A Shape is the base class for representing a scalable geometrical shape. The shapes have different
+ways of defining how they look, but they all have a size and angle that can be modified and the
+shape definition will scale or rotate the shape accordingly.
+
+There are currently three shapes: [Polygon}(#Polygon), [Rectangle](#Rectangle) and
+[Cirlce](#Circle).
```

Comment:
Rectangle is just a simpler (and more efficient) way to create an axis-aligned rectangle instead of using Polygon.

Reply:
That is explained in the rectangle section I think

Reply:
yes. also it does not need to be AA, I later discovered